include/revng/ADT/RecursiveCoroutine-coroutine.h

```diff
@@ -181,9 +182,9 @@ struct RecursiveCoroutine<ReturnT>::promise_type : public PromiseBase {
   std::optional<ReturnT> CurrValue = std::nullopt;
 };
 
-template<typename CoroutineT, typename... Args>
-auto rc_run(CoroutineT F, Args... Arguments) {
-  return F(Arguments...).run();
+template<typename CoroutineT, typename... FormalArgs, typename... ActualArgs>
```

Comment:
why is it necessary to take FormalArgs as template arguments?

```
template<typename CoroutineT, typename... ActualArgs>
auto rc_run(CoroutineT F, ActualArgs &&... args) {
￼  return F(std::forward<ActualArgs>(args)...).run();
￼}
```
should be enough?

Reply:
Good point! It was just a leftover. I removed it.